app/_components/dropdownList/DropdownList.tsx

```diff
@@ -0,0 +1,50 @@
+"use client"
+
+import clsx from "clsx"
+import { DropdownMenu, DropdownMenuContent, DropdownMenuItem, DropdownMenuTrigger } from "components/ui/dropdown-menu"
+import ChevronDownIcon from "../svgs/ChevronDownIcon"
+
+interface Item {
+  id: number
+  title?: string
+}
+
+interface DropdownListProps {
+  className?: string
+  itemClassName?: string
+  list: Item[]
+  title: string
+  selectedValue?: number
+  onSelect: (id: number) => void
+  enabled?: boolean
+}
+
+export function DropdownList({
+  list,
+  className,
+  itemClassName = "w-full",
+  title,
+  selectedValue,
+  onSelect,
+  enabled = true,
+}: DropdownListProps) {
+  const selectedTitle =
+    selectedValue !== undefined ? list.find((item) => item.id === selectedValue)?.title || title : title
+
```

Comment:
lets make it clearer 
const selectedItem = selectedValue == undefined  ? selectedValue : list.find((item) => item.id === selectedValue)
const selectedTitle = selectedItem ? selectedItem.title : title